pyogrio/geopandas.py

```diff
@@ -331,9 +331,15 @@ def read_dataframe(
 
         del table
 
-        for ogr_subtype, c in zip(meta["ogr_subtypes"], df.columns):
+        for ogr_subtype, c in zip(meta["ogr_subtypes"], meta["fields"]):
             if ogr_subtype == "OFSTJSON":
-                df[c] = df[c].map(json.loads, na_action="ignore")
+                # When reading .parquet files with arrow, JSON fields are already
+                # parsed, so only parse if strings.
+                dtype = pd.api.types.infer_dtype(df[c])
+                if dtype == "string":
+                    df[c] = df[c].map(json.loads, na_action="ignore")
```

Comment:
I would maybe also in general add a try/except around this (and potentially raise a warning if it fails), so we can read in "invalid" data gracefully